lib/fprotect/fprotect_acl.c

```diff
@@ -32,3 +33,30 @@ int fprotect_area(uint32_t start, size_t length)
 
 	return 0;
 }
+
+#if defined(CONFIG_FPROTECT_ENABLE_NO_ACCESS)
+
+int fprotect_area_no_access(uint32_t start, size_t length)
+{
+
+	__ASSERT_NO_MSG(nrf_ficr_codepagesize_get(NRF_FICR) ==
+			CONFIG_FPROTECT_BLOCK_SIZE);
+
+	if (region_idx >= ACL_REGIONS_COUNT) {
+		return -ENOSPC;
+	}
+
+	if (start  % nrf_ficr_codepagesize_get(NRF_FICR) != 0 ||
```

Comment:
Is this an exact duplicate of the logic in `fprotect_area`? If so, should we pull it out as a separate function to avoid duplication?

Reply:
In fact, this whole function seems a bit like a duplicate. Could we create a local function which takes the `ACL_PERM` as an argument, and then call that from the two public functions?

Reply:
Done :)